src/lib.rs

```diff
@@ -379,16 +394,70 @@ impl GvasFile {
     ///     "Guid".to_string(),
     /// );
     ///
-    /// let gvas_file = GvasFile::read_with_hints(&mut file, &hints);
+    /// let gvas_file = GvasFile::read_with_hints(&mut file, GameVersion::Default, &hints);
     ///
     /// println!("{:#?}", gvas_file);
     /// # Ok::<(), Error>(())
     /// ```
     pub fn read_with_hints<R: Read + Seek>(
         cursor: &mut R,
+        game_version: GameVersion,
         hints: &HashMap<String, String>,
     ) -> Result<Self, Error> {
-        let header = GvasHeader::read(cursor)?;
+        let deserialized_game_version: DeserializedGameVersion;
+        let mut cursor = match game_version {
+            GameVersion::Default => {
+                deserialized_game_version = DeserializedGameVersion::Default;
+                let mut data = Vec::new();
+                cursor.read_to_end(&mut data)?;
+                Cursor::new(data)
```

Comment:
I think this can be simplified:

```suggestion
                cursor
```

Reply:
This would return a mutable reference to cursor instead of a value so the type checker would error